tests/st/policy/test_felix_gateway.py

```diff
@@ -0,0 +1,510 @@
+        })
+        retry_until_success(self.assert_host_can_not_curl_ext, 3)
+
+    def add_policy(self, policy_data):
+        self._apply_resources(policy_data, self.gateway)
+
+    def add_gateway_internal_iface(self):
+        host_endpoint_data = {
+            'apiVersion': 'v1',
+            'kind': 'hostEndpoint',
+            'metadata': {
+                'name': 'gw-int',
+                'node': self.gateway_hostname,
+                'labels': {'role': 'gateway-int'}
+            },
+            'spec': {
+                'interfaceName': 'eth0'
+            }
+        }
+        self._apply_resources(host_endpoint_data, self.gateway)
+
+    def add_gateway_external_iface(self):
+        host_endpoint_data = {
+            'apiVersion': 'v1',
+            'kind': 'hostEndpoint',
+            'metadata': {
+                'name': 'gw-ext',
+                'node': self.gateway_hostname,
+                'labels': {'role': 'gateway-ext'}
+            },
+            'spec': {
+                'interfaceName': 'eth1'
+            }
+        }
+        self._apply_resources(host_endpoint_data, self.gateway)
+
+    def add_host_iface(self):
+        host_endpoint_data = {
+            'apiVersion': 'v1',
+            'kind': 'hostEndpoint',
+            'metadata': {
+                'name': 'host-int',
+                'node': self.host_hostname,
+                'labels': {'role': 'host'}
+            },
+            'spec': {
+                'interfaceName': 'eth0',
+                'expectedIPs': [self.get_container_ip('cali-st-host')],
+            }
+        }
+        self._apply_resources(host_endpoint_data, self.gateway)
+
+    @staticmethod
+    def _apply_resources(resources, host):
+        # Apply new resources.
+        host.writefile("resources",
+                       yaml.dump(resources, default_flow_style=False))
+        host.calicoctl("apply -f resources")
+
+    def assert_host_can_curl_ext(self):
+        try:
+            self.host.execute("curl --fail -m 1 -o /tmp/nginx-index.html %s" % self.ext_server_ip)
+        except subprocess.CalledProcessError:
+            _log.exception("Internal host failed to curl external server IP: %s",
+                           self.ext_server_ip)
+            self.fail("Internal host failed to curl external server IP: %s" % self.ext_server_ip)
+
+    def assert_host_can_not_curl_ext(self):
+        try:
+            self.host.execute("curl --fail -m 1 -o /tmp/nginx-index.html %s" % self.ext_server_ip)
+        except subprocess.CalledProcessError:
+            return
+        else:
+            self.fail("Internal host can curl external server IP: %s" % self.ext_server_ip)
+
+    def remove_pol_and_endpoints(self):
+        self.delete_all("pol")
+        self.delete_all("hostEndpoint")
+        # Wait for felix to remove the policy and allow traffic through the gateway.
+        retry_until_success(self.assert_host_can_curl_ext)
+
+    def delete_all(self, resource):
```

Comment:
We should move this into the FV libraries.  Not necessarily something you should do for this PR, but someone, soon.